src/main/io/gps.c

```diff
@@ -1368,68 +1408,76 @@ void gpsUpdate(timeUs_t currentTimeUs)
             isFast = false;
             rescheduleTask(TASK_SELF, TASK_PERIOD_HZ(TASK_GPS_RATE));
         }
-    } else if (gpsConfig()->provider == GPS_MSP) {
+        break;
+
+    case GPS_MSP:
         if (GPS_update & GPS_MSP_UPDATE) { // GPS data received via MSP
-            if (gpsData.state == GPS_STATE_INITIALIZED) {
-                gpsSetState(GPS_STATE_RECEIVING_DATA);
-            }
+        if (gpsData.state == GPS_STATE_INITIALIZED) {
```

Comment:
It seems that indent is work here (it is inside if (GPS_update & GPS_MSP_UPDATE) )